lib/pages/my_home_page.dart:56:5: Error: Type 'Reader' not found

win10,android studio 安卓环境报错，请教如何处理

`Launching lib\main.dart on sdk gphone x86 in debug mode...
Running Gradle task 'assembleDebug'...
lib/pages/my_home_page.dart:56:5: Error: Type 'Reader' not found.
    Reader read,
    ^^^^^^
lib/pages/my_home_page.dart:56:5: Error: 'Reader' isn't a type.
    Reader read,
    ^^^^^^
lib/providers/ncnn_yolox_controller.dart:22:9: Error: Type 'Reader' not found.
  final Reader _read;
        ^^^^^^
lib/providers/ncnn_yolox_controller.dart:22:9: Error: 'Reader' isn't a type.
  final Reader _read;
        ^^^^^^
lib/providers/my_camera_controller.dart:14:9: Error: Type 'Reader' not found.
  final Reader _read;
        ^^^^^^
lib/providers/my_camera_controller.dart:14:9: Error: 'Reader' isn't a type.
  final Reader _read;
        ^^^^^^


FAILURE: Build failed with an exception.`

--- a/example/lib/pages/my_home_page.dart
+++ b/example/lib/pages/my_home_page.dart
@@ -22,7 +22,7 @@ class MyHomePage extends HookConsumerWidget {
             title: const Text('from gallery image'),
             onTap: () => _pickImage(
               context,
-              ref.read,
+              ref.read(ncnnYoloxController.notifier),
               ImageSource.gallery,
             ),
           ),
@@ -30,7 +30,7 @@ class MyHomePage extends HookConsumerWidget {
             title: const Text('from camera image'),
             onTap: () => _pickImage(
               context,
-              ref.read,
+              ref.read(ncnnYoloxController.notifier),
               ImageSource.camera,
             ),
           ),
@@ -53,14 +53,13 @@ class MyHomePage extends HookConsumerWidget {
 
   Future<void> _pickImage(
     BuildContext context,
-    Reader read,
+    NcnnYoloxController controller,
     ImageSource imageSource,
   ) async {
     final navigator = Navigator.of(context);
 
     final file = await ImagePicker().pickImage(source: imageSource);
     if (file != null) {
-      final controller = read(ncnnYoloxController.notifier);
       await controller.initialize();
       await controller.detectFromImageFile(file);
 

--- a/example/lib/providers/my_camera_controller.dart
+++ b/example/lib/providers/my_camera_controller.dart
@@ -5,13 +5,13 @@ import 'package:ncnn_yolox_flutter/ncnn_yolox_flutter.dart';
 import 'package:ncnn_yolox_flutter_example/providers/ncnn_yolox_controller.dart';
 
 final myCameraController = Provider(
-  (ref) => MyCameraController(ref.read),
+  MyCameraController.new,
 );
 
 class MyCameraController {
-  MyCameraController(this._read);
+  MyCameraController(this.ref);
 
-  final Reader _read;
+  final Ref ref;
 
   CameraController? _cameraController;
 
@@ -25,7 +25,7 @@ class MyCameraController {
   bool _isProcessing = false;
 
   Future<void> startImageStream() async {
-    await _read(ncnnYoloxController.notifier).initialize();
+    await ref.read(ncnnYoloxController.notifier).initialize();
 
     final camera = (await availableCameras())[0];
 
@@ -43,7 +43,9 @@ class MyCameraController {
         }
 
         _isProcessing = true;
-        await _read(ncnnYoloxController.notifier).detectFromCameraImage(image);
+        await ref
+            .read(ncnnYoloxController.notifier)
+            .detectFromCameraImage(image);
         _isProcessing = false;
       },
     );

--- a/example/lib/providers/ncnn_yolox_controller.dart
+++ b/example/lib/providers/ncnn_yolox_controller.dart
@@ -13,13 +13,13 @@ import 'package:ncnn_yolox_flutter_example/providers/ncnn_yolox_options.dart';
 
 final ncnnYoloxController =
     StateNotifierProvider<NcnnYoloxController, List<YoloxResults>>(
-  (ref) => NcnnYoloxController(ref.read),
+  NcnnYoloxController.new,
 );
 
 class NcnnYoloxController extends StateNotifier<List<YoloxResults>> {
-  NcnnYoloxController(this._read) : super([]);
+  NcnnYoloxController(this.ref) : super([]);
 
-  final Reader _read;
+  final Ref ref;
 
   final _ncnnYolox = NcnnYolox();
 
@@ -31,10 +31,10 @@ class NcnnYoloxController extends StateNotifier<List<YoloxResults>> {
     await _ncnnYolox.initYolox(
       modelPath: 'assets/yolox/yolox.bin',
       paramPath: 'assets/yolox/yolox.param',
-      autoDispose: _read(ncnnYoloxOptions).autoDispose,
-      nmsThresh: _read(ncnnYoloxOptions).nmsThresh,
-      confThresh: _read(ncnnYoloxOptions).confThresh,
-      targetSize: _read(ncnnYoloxOptions).targetSize,
+      autoDispose: ref.read(ncnnYoloxOptions).autoDispose,
+      nmsThresh: ref.read(ncnnYoloxOptions).nmsThresh,
+      confThresh: ref.read(ncnnYoloxOptions).confThresh,
+      targetSize: ref.read(ncnnYoloxOptions).targetSize,
     );
   }
 
@@ -47,7 +47,7 @@ class NcnnYoloxController extends StateNotifier<List<YoloxResults>> {
         file.path,
       ).readAsBytesSync(),
     );
-    _read(previewImage.state).state = decodedImage;
+    ref.read(previewImage.notifier).state = decodedImage;
   }
 
   Future<void> detectFromCameraImage(CameraImage cameraImage) async {
@@ -67,10 +67,10 @@ class NcnnYoloxController extends StateNotifier<List<YoloxResults>> {
               v: cameraImage.planes[2].bytes,
               height: cameraImage.height,
               deviceOrientationType:
-                  _read(myCameraController).deviceOrientationType,
-              sensorOrientation: _read(myCameraController).sensorOrientation,
+                  ref.read(myCameraController).deviceOrientationType,
+              sensorOrientation: ref.read(myCameraController).sensorOrientation,
               onDecodeImage: (image) {
-                _read(previewImage.state).state = image;
+                ref.read(previewImage.notifier).state = image;
                 completer.complete();
               },
             )
@@ -82,10 +82,10 @@ class NcnnYoloxController extends StateNotifier<List<YoloxResults>> {
               pixels: cameraImage.planes[0].bytes,
               height: cameraImage.height,
               deviceOrientationType:
-                  _read(myCameraController).deviceOrientationType,
-              sensorOrientation: _read(myCameraController).sensorOrientation,
+                  ref.read(myCameraController).deviceOrientationType,
+              sensorOrientation: ref.read(myCameraController).sensorOrientation,
               onDecodeImage: (image) {
-                _read(previewImage.state).state = image;
+                ref.read(previewImage.notifier).state = image;
                 completer.complete();
               },
             )

--- a/example/pubspec.yaml
+++ b/example/pubspec.yaml
@@ -12,7 +12,7 @@ dependencies:
   flutter:
     sdk: flutter
   freezed_annotation: ^2.2.0
-  hooks_riverpod: ^2.0.0-dev.9
+  hooks_riverpod: ^2.1.1
   image: ^3.2.2
   image_picker: ^0.8.6
   ncnn_yolox_flutter:
@@ -22,7 +22,7 @@ dev_dependencies:
   build_runner: ^2.3.2
   flutter_test:
     sdk: flutter
-  freezed: ^2.2.1
+  freezed: ^2.3.0
   pedantic_mono: ^1.20.0
 
 flutter:

